src/query_param.c

```diff
@@ -28,11 +28,10 @@ QueryParam *NewGeoFilterQueryParam_WithParams(struct QueryParseCtx *q, QueryToke
   assert (unit->type != QT_TERM_CASE);
   if (unit->type == QT_TERM && unit->s) {
     gf->unitType = GeoDistance_Parse_Buffer(unit->s, unit->len);
-  } else if (unit->type == QT_PARAM_GEO_UNIT) {
-    QueryParam_SetParam(q, &ret->params[3], &gf->unitType, NULL, unit);
+  } else if (unit->type == QT_NUMERIC || unit->type == QT_SIZE) {
+    QERR_MKSYNTAXERR(q->status, "Invalid GeoFilter unit");
   } else {
-    // Invalid unit->type
-    gf->unitType = GeoDistance_Parse_Buffer("", 0);
+    QueryParam_SetParam(q, &ret->params[3], &gf->unitType, NULL, unit);
```

Comment:
This isn't equivalent. If one day someone will add some other type to be valid, it will default to a valid GeoFilter node instead of getting an error.
I think the `else` clause should throw the error, we should just add a `else if (unit->type == QT_NUMERIC || unit->type == QT_SIZE) { throw error }` block before the `else` since these are the new cases that we introduced in this PR.

Reply:
I modified the code to handle only the valid types `QT_TERM` and `QT_PARAM_GEO_UNIT`, and throw error if a different type is received. If another type is added later, a new comparison should be added to avoid the error. WDYT?